test/idler_assertion.go

```diff
@@ -266,3 +273,23 @@ func (a *IdleablePayloadAssertion) StatefulSetScaledUp(statefulSet *appsv1.State
 	assert.Equal(a.t, int32(3), *s.Spec.Replicas)
 	return a
 }
+
+func (a *IdleablePayloadAssertion) VMRunning(vm *unstructured.Unstructured) *IdleablePayloadAssertion {
+	vm, err := a.dynamicClient.Resource(vmGVR).Namespace(vm.GetNamespace()).Get(context.TODO(), vm.GetName(), metav1.GetOptions{})
+	require.NoError(a.t, err)
+	val, found, err := unstructured.NestedBool(vm.Object, "spec", "running")
+	require.NoError(a.t, err)
+	assert.True(a.t, found)
+	assert.True(a.t, val)
+	return a
+}
+
+func (a *IdleablePayloadAssertion) VMStopped(vm *unstructured.Unstructured) *IdleablePayloadAssertion {
+	vm, err := a.dynamicClient.Resource(vmGVR).Namespace(vm.GetNamespace()).Get(context.TODO(), vm.GetName(), metav1.GetOptions{})
+	require.NoError(a.t, err)
+	val, found, err := unstructured.NestedBool(vm.Object, "spec", "running")
+	require.NoError(a.t, err)
+	assert.True(a.t, found)
+	assert.False(a.t, val)
+	return a
+}
```

Comment:
Or something like this:

```suggestion
func (a *IdleablePayloadAssertion) VMRunning(vm *unstructured.Unstructured) *IdleablePayloadAssertion {
	return a.vmRunning(vm, true)
}

func (a *IdleablePayloadAssertion) VMStopped(vm *unstructured.Unstructured) *IdleablePayloadAssertion {
	return a.vmRunning(vm, false)
}

func (a *IdleablePayloadAssertion) vmRunning(vm *unstructured.Unstructured, running bool) *IdleablePayloadAssertion {
	vm, err := a.dynamicClient.Resource(vmGVR).Namespace(vm.GetNamespace()).Get(context.TODO(), vm.GetName(), metav1.GetOptions{})
	require.NoError(a.t, err)
	val, found, err := unstructured.NestedBool(vm.Object, "spec", "running")
	require.NoError(a.t, err)
	assert.True(a.t, found)
	assert.Equal(a.t, running, val)
	return a
}
```

Reply:
Yeah, I actually did initially have a function `vmRunning(vm *unstructured.Unstructured, running bool)` but liked `VMRunning` and `VMStopped` from the test perspective. This is a good compromise. 👍